File: LeetCode/[M] - 841. Keys and Rooms/solution.go
package main

func canVisitAllRooms(rooms [][]int) bool {
	marked := make(map[int]struct{})
	marked[0] = struct{}{}
	stack := []int{0}
	for len(stack) != 0 {
		v := stack[0]
		stack = stack[1:]
		for _, w := range rooms[v] {
			if _, ok := marked[w]; !ok {
				stack = append(stack, w)
				marked[w] = struct{}{}
			}
		}
	}
	return len(marked) == len(rooms)
}

File: LeetCode/[M] - 841. Keys and Rooms/solution.py
from typing import List


class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        marked = set([0])
        stack = [0]
        while stack:
            v = stack.pop()
            for w in rooms[v]:
                if w not in marked:
                    stack.append(w)
                    marked.add(w)
        return len(marked) == len(rooms)
